feature(artifacts-test): support to test suse linux

amoskong opened the pull request:
ticket: https://trello.com/c/oxsOLpBg/3489-8277-it-is-now-possible-to-install-scylla-on-suse-linux-enterprise-server-using-the-standard-rpm-packages

Currently we only want to support one release for Suse Enterprise Linux.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

amoskong commented:
Note: One scylla fix needs to be backported to 4.5

- https://github.com/scylladb/scylla/issues/9045
- https://github.com/scylladb/scylla-tools-java/commit/4ef8049e0717a2ea7998a21ff8529ae293a37a6c

amoskong commented:
Current status:

Takuya already fixed the _yaml.so dependency issue in scylla-tools-java.

Currently we need to enable Legacy and Python_2 modules for resolving the python2-PyYAML and java-1.8 dependency.
But it doesn't work all the time (it works sometimes), I'm working to address the problem.

```
scylla-test@artifacts-sles15-jenkins-db-node-ee0cda78-0-1:~> sudo zypper mr -e Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Pool
Repository Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Pool not found.
```

amoskong commented:
I had resolved the dependency issue in the PR, moving the PR to ready state.

Now the test failed for scylla issues:

- https://github.com/scylladb/scylla/issues/9172
- https://github.com/scylladb/seastar/issues/930  (posted a fix to seastar)

roydahan commented on sdcm/cluster.py:
```diff
@@ -2003,6 +2036,19 @@ def install_scylla(self, scylla_repo):
+            # self.remoter.sudo('zypper mr -e Legacy_Module_x86_64:SLE-Module-Legacy15-SP3-Pool')
+            self.remoter.sudo('SUSEConnect --product sle-module-legacy/15.3/x86_64')
+            self.remoter.sudo('SUSEConnect --product sle-module-python2/15.3/x86_64')
+            self.remoter.sudo('zypper install -y python2-PyYAML', ignore_status=True)
+            self.remoter.sudo('zypper install -y python3-PyYAML', ignore_status=True)
```
Is this part of the official documentation?

amoskong replied:
We don't have install guide for Suse now.

roydahan reviewed (commented):
All in all looks good to me, but AFAIU it doesn't work yet, right?

Is the test synced with the official documentation?
Can you please link here the documentation?

dkropachev commented on sdcm/cluster.py:
```diff
@@ -2292,7 +2345,8 @@ def install_mgmt(self, scylla_mgmt_repo: str, package_url: Optional[str] = None)
 
         if self.distro.is_rhel_like:
             self.install_epel()
-            self.remoter.sudo("yum install python36-PyYAML -y", retry=3)
```
Why this one was removed ?

amoskong replied:
added back, thanks

enaydanov reviewed (changes requested):
Please, add unit test for `Distro.is_sles` and `Distro.is_sles15` to `unit_tests/test_utils_distro.py`

```
test@sles15:~> cat /etc/os-release
NAME="SLES"
VERSION="15-SP3"
VERSION_ID="15.3"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP3"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp3"
DOCUMENTATION_URL="https://documentation.suse.com/"
test@sles15:~>
```

amoskong commented:
Rebased the PR to latest master and added unittest.

The test sill failed for scylla issue, I just pinged in the fix PR:
- test job: https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-sles15-test/12/console
- seastar issue: https://github.com/scylladb/seastar/issues/930
- fix: https://github.com/scylladb/seastar/pull/931

We can consider to merge this PR.

enaydanov commented on sdcm/cluster.py:
```diff
@@ -1981,6 +2008,10 @@ def update_repo_cache(self):
                 self.remoter.run('sudo rm -rf /var/cache/yum/')
                 self.remoter.run('sudo yum makecache', retry=3)
+            elif self.distro.is_sles:
+                self.remoter.run('sudo zypper clean all')
+                self.remoter.run('sudo rm -rf /var/cache/zypp/')
+                self.remoter.run('sudo zypper refresh', retry=3)
```
Please, use `self.runner.sudo()` here

amoskong replied:
fixed.

enaydanov commented on sdcm/cluster.py:
```diff
@@ -1997,6 +2028,8 @@ def upgrade_system(self):
             else:
                 self.remoter.run('sudo yum update -y --skip-broken', retry=3)
+        elif self.distro.is_sles:
+            self.remoter.run('sudo zypper update -y -l')
```
Please, use `self.runner.sudo()` here

enaydanov reviewed (changes requested):
Except run/sudo stuff I'm OK with these changes.

amoskong commented on sdcm/cluster.py:
```diff
@@ -1855,6 +1863,13 @@ def download_scylla_repo(self, scylla_repo):
+            repo_path = '/etc/zypp/repos.d/scylla.repo'
+            self.remoter.run('sudo curl -o %s -L %s' % (repo_path, scylla_repo))
+            self.remoter.run('sudo chown root:root %s' % repo_path)
+            self.remoter.run('sudo chmod 644 %s' % repo_path)
```
fixed.

roydahan commented:
@amoskong can you please check with @lauranovich there is a matching documentation?
